opentelemetry/src/main/java/io/grpc/opentelemetry/OpenTelemetryTracingModule.java

```diff
@@ -315,9 +316,59 @@ final class TracingServerSpanPropagationInterceptor implements ServerInterceptor
     public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call,
         Metadata headers, ServerCallHandler<ReqT, RespT> next) {
       Span span = otelSpan.get(io.grpc.Context.current());
-      try (Scope scope = Context.current().with(span).makeCurrent()) {
+      if (span == null) {
+        logger.log(Level.FINE, "Server span not found. ServerTracerFactory for server "
+            + "tracing must be set.");
         return next.startCall(call, headers);
       }
+      try (Scope scope = Context.current().with(span).makeCurrent()) {
```

Comment:
Do the `Context.current().with(span)` once in this method, then call `makeCurrent()` on that single instance in all the places. The context shouldn't be changing between the callbacks.